gcloud/pubsub/_gax.py

```diff
@@ -0,0 +1,173 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+try:
+    from google.gax import CallOptions
+except ImportError:
+    _HAVE_GAX = False
+else:
+    _HAVE_GAX = True
+
+
+    from google.gax.errors import GaxError
+    from google.pubsub.v1.pubsub_pb2 import PubsubMessage
+
+    from gcloud.exceptions import Conflict
+    from gcloud.exceptions import NotFound
+
+
+    class _PublisherAPI(object):
+
+        def __init__(self, gax_api):
+            self._gax_api = gax_api
+
+        def list_topics(self, project):
+            """List topics for the project associated with this API.
+
+            See:
+            https://cloud.google.com/pubsub/reference/rest/v1/projects.topics/list
+
+            :type project: string
+            :param project: project ID
+
+            :rtype: tuple, (list, str)
+            :returns: list of ``Topic`` resource dicts, plus a
+                    "next page token" string:  if not None, indicates that
+                    more topics can be retrieved with another call (pass that
+                    value as ``page_token``).
+            """
+            options = CallOptions(is_page_streaming=False)
+            path = 'projects/%s' % (project,)
+            response = self._gax_api.list_topics(path, options)
+            topics = [{'name': topic_pb.name} for topic_pb in response.topics]
```

Comment:
All the `from_api_repr` methods act on the dict resources returned by the APIs (that is what the name means).  It just so happens that (in this version of the Pubsub API) the `Topic` resource is a single-key mapping .  `Subscription`, for instance, has a [much more complicated `from_api_repr`](https://github.com/GoogleCloudPlatform/gcloud-python/blob/77123bb911f939b80c6d132d8cc12e6a16517539/gcloud/pubsub/subscription.py#L72-L109).

Keeping to a  "standard" protocol between "transport helpers" and the domain objects which consume the resources they produce makes the whole system easier to reason about:  making `Topic.from_api_repr` a special case to avoid the miniscule performance impact of creating a dict damages that quality.

if https://github.com/google/protobuf/issues/1351 were resolved, I would just update this code to use `protobuf.json_format.MessageToJsonObject` instead of creating the dict by hand.